src/GitHub.App/Factories/ModelServiceFactory.cs

```diff
@@ -0,0 +1,64 @@
+﻿using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Threading;
+using System.Threading.Tasks;
+using GitHub.Models;
+using GitHub.Services;
+using Microsoft.VisualStudio.Shell;
+
+namespace GitHub.Factories
+{
+    [Export(typeof(IModelServiceFactory))]
+    [PartCreationPolicy(CreationPolicy.Shared)]
+    public sealed class ModelServiceFactory : IModelServiceFactory, IDisposable
+    {
+        readonly IApiClientFactory apiClientFactory;
+        readonly IHostCacheFactory hostCacheFactory;
+        readonly IAvatarProvider avatarProvider;
+        readonly Dictionary<IConnection, ModelService> cache = new Dictionary<IConnection, ModelService>();
+        readonly SemaphoreSlim cacheLock = new SemaphoreSlim(1);
+
+        [ImportingConstructor]
+        public ModelServiceFactory(
+            IApiClientFactory apiClientFactory,
```

Comment:
It would be better to use `ISimpleApiClientFactory` here, so we can slowly move away from `IApiClientFactory` and get rid of it.

Reply:
Yes, but `ModelService` still relies on `IApiClient` rather than `ISimpleApiClient` and needs the reactive stuff, so if we want to make that change it will have to be later I think.